pkg/infrastructure/nutanix/clusterapi/clusterapi.go

```diff
@@ -146,44 +147,86 @@ func (p Provider) Ignition(ctx context.Context, in infracapi.IgnitionInput) ([]b
 	}
 	imgReq.Metadata = imgMeta
 
+	retries := 5
+	completed := false
+	for i := 0; !completed && i < retries; i++ {
+		imgUUID, err := createImage(ctx, nutanixCl, imgReq, imgName)
+		if err != nil {
+			logrus.Errorf("failed to create image object %s: %v (retries: %v)", imgName, err, i)
+			if imgUUID != "" {
+				if err = deleteImage(ctx, nutanixCl, imgUUID); err != nil {
+					logrus.Errorf("failed to delete image object %s (uuid: %s): %v", imgName, imgUUID, err)
+				}
+			}
+			continue
+		}
+		logrus.Infof("Created the image object %s (uuid: %s)", imgName, imgUUID)
+
+		err = uploadImageData(ctx, nutanixCl, imgName, imgUUID, imgPath)
+		if err != nil {
+			logrus.Errorf("failed to upload image data for %s (uuid: %s): %v (retries: %v)", imgName, imgUUID, err, i)
+			if err = deleteImage(ctx, nutanixCl, imgUUID); err != nil {
+				logrus.Errorf("failed to delete image object %s (uuid: %s): %v", imgName, imgUUID, err)
+			}
+		} else {
+			completed = true
+		}
+	}
```

Comment:
Instead of (create or delete) + (upload or delete) in a loop, wouldn't it be better to separate into 2 retriables? That way you just create (and delete, in the failure path) the image once.
1. A loop (wait.ExponentialBackoffWithContext) just to create the image. Return failure if it didn't succeed
2. A loop (wait.ExponentialBackoffWithContext) just to upload the image. Delete image and return failure if it didn't succeed.

Reply:
Made the change as suggested.